src/daemon/distribution.c

```diff
@@ -0,0 +1,205 @@
+//
+// Created by Svetlana Shmidt on 24.07.2020.
+//
+
+#include "distribution.h"
+
+#include <errno.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+
+typedef struct bucket_s {
+  uint64_t bucket_counter;
+  double minimum, maximum;
+} bucket_t;
+
+struct distribution_s {
+  bucket_t *tree;
+  size_t num_buckets;
+  double total_sum;
+};
+
+size_t left_child_index(size_t node_index, size_t left, size_t right) {
+  return node_index + 1;
+}
+
+size_t right_child_index(size_t node_index, size_t left, size_t right) {
```

Comment:
Can you please add a comment here?

Typically, you want to explain **why** in a comment, not **what**. "Calculates the index of the right child" just re-states what the function name already says, i.e. it explains *what* is happening, which is not that useful.

Why is `node_index + 2*(mid - left + 1)` the index of the right child? An answer could possibly be structured like this: "This code uses an Euler path to avoid gaps in the tree-to-array mapping. The left subtree has {so many} nodes, therefore the right subtree starts at {that value}. For a detailed explanation, see {link}".